src/containers/Consulting/Consulting.tsx

```diff
@@ -0,0 +1,232 @@
+  const [when, setWhen] = useState<any>(new Date());
+  const [duration, setDuration] = useState<number>();
+  const [content, setContent] = useState('');
+  const [isBillable, setIsBillable] = useState<any>(null);
+  const [organization, setOrganization] = useState<any>(null);
+
+  const states = {
+    participants,
+    staff,
+    when,
+    duration,
+    content,
+    isBillable,
+    organization,
+  };
+
+  const { data: organizationList } = useQuery(FILTER_ORGANIZATIONS, {
+    variables: setVariables(),
+  });
+
+  if (!organizationList) {
+    return <Loading />;
+  }
+
+  const setStates = ({
+    participants: supportMembers,
+    staff: staffMembers,
+    when: consultingDate,
+    duration: durationInMin,
+    content: description,
+    isBillable: billable,
+    organization: org,
+    organizationName,
+  }: any) => {
+    setParticipants(supportMembers);
+    setStaff(staffMembers);
+    setWhen(consultingDate);
+    setDuration(durationInMin);
+    setContent(description);
+    setIsBillable(billable);
+    /**
+     * When edit organization is undefined
+     * since we are getting organizationName,
+     * we need to get object from organizationList for selected organizationName
+     */
+    if (!org) {
+      const selectedOrg = organizationList.organizations.find(
+        ({ name }: { name: string }) => name === organizationName
+      );
+      setOrganization(selectedOrg);
+    } else setOrganization(org);
+  };
+
+  const FormSchema = Yup.object().shape({
+    participants: Yup.string().required(t('Participants are required.')),
+    staff: Yup.string().required(t('NGO staff members are required.')),
+    content: Yup.string().required(t('Description is required')),
+    when: Yup.date().nullable().required(t('Date is required and should be valid.')),
+    duration: Yup.number()
+      .nullable()
+      .moreThan(0, 'Duration should be greater than 0')
+      .required(t('Duration is required')),
+    isBillable: Yup.boolean().nullable().required('Required'),
+    organization: Yup.object().nullable().required('Organization is required'),
+  });
+
+  const queries: any = {
+    getItemQuery: GET_CONSULTING_HOURS_BY_ID,
+    createItemQuery: CREATE_CONSULTING_HOUR,
+    updateItemQuery: UPDATE_CONSULTING_HOURS,
+    deleteItemQuery: DELETE_CONSULTING_HOURS,
+  };
+
+  const formFields = (organizationOptions: Array<any>) => [
+    {
+      component: AutoComplete,
+      name: 'organization',
+      placeholder: t('Select Organization'),
+      options: organizationOptions,
+      optionLabel: 'name',
+      multiple: false,
+      textFieldProps: {
+        label: t('Select Organization'),
+        variant: 'outlined',
+      },
+      onChange: (val: any) => setOrganization(val),
+    },
+    {
+      component: Input,
+      name: 'participants',
+      type: 'text',
+      placeholder: t('Participants'),
+      inputProp: {
+        onChange: (event: any) => setParticipants(event.target.value),
```

Comment:
Do we need this listener here? Can you also check if these are needed in others as well because formik handles it internally?

Reply:
Yes, we need listeners because we want to have both formik and local states synced. Otherwise, values for formik state reset with our local state(since local state values are not being updated) when you change the dropdown value.